tests/Core/Net/WhoisClientTest.php

```
@@ -0,0 +1,193 @@
+    }
+}
+
+/** A chain slower than the budget allows. */
+final class SlowWhoisClient extends WhoisClient
+{
+    public function __construct(private float $budget)
+    {
+        parent::__construct($budget);
+    }
+
+    protected function ask(string $server, string $query, float $deadline): ?string
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$server'. (undefined)

(UnusedFormalParameter)